lib/payment_dupes.rb

```diff
@@ -43,36 +47,65 @@ def self.can_copy_designation?(src, target, designations_to_become_comments)
         true
     end
 
-    def self.copy_comment(source, target)
+    def self.copy_comment(source, target, designations_to_become_comments)
         return true if source.donation.comment.blank?
         return true if target.donation.comment.present? && (source.donation.comment.blank? || target.donation.comment == source.donation.comment)
-        return false if target.donation.comment.present?
+        if target.donation.comment.present? 
+            if designations_to_become_comments.any? { |d| target.donation.comment.include?(d) }
+                designations_already_copied_to_comment = designations_to_become_comments.select { |d| target.donation.comment.include?(d) }
+                comment = target.donation.comment
+                designations_already_copied_to_comment.each do |d|
+                    comment = target.donation.comment.gsub(" \nDesignation: #{d}", "")
+                    comment = target.donation.comment.gsub("Designation: #{d}", "")
```

Comment:
Why are these both here?

Reply:
Because when a donation doesn't have a comment, we just copy the designation without a \n, but if it has a comment already, we copy with a \n. Those two lines make sure we're removing both cases. See: https://github.com/CommitChange/houdini/pull/268/files#diff-0dbca3d4691fa54ac9678cda39b5ec2e9d18eed195ffa425fe2adbed140d82d5R23-R27

Reply:
Alright.